main.js

```diff
@@ -1,53 +1,73 @@
 // ------------- part 6 ----------------
 
-let user = 'John Doe';
-console.log(user);
+palindrome = word => {
+    let lastLater = word.length - 1;
+
+    for (var i = 0; i < word.length; i++) {
+        if (word.charAt(i) !== word.charAt(lastLater))
+            return 0;
+
+        lastLater--;
+
+        if (word.length % 2 === 0) {
+            if (i === (word.length / 2)) 
+                return 1; 
+        } else {
+            if (i === (word.length / 2) - 0.5) 
+                return 1;
+        }
+    }    
+}
+
+result1 = name => console.log(palindrome(name));
 
-let student = 'Pavel';
-console.log(student);
+result1('шалаш');
+result1('анна');
+
+result1('телефон');
+result1('часы');
 
-user = student;
-console.log(user);
 
 // ------------- part 7 ----------------
 
-let test = 1;
-test++;
 
-test += '1';
-console.log(test);
+min = (a, b) => (a < b ? a : b < a ? b : 'Значения равны!');
+max = (a, y) => (a > y ? a : y > a ? y : 'Значения равны!');
 
-test -= 1;
-console.log(test);
+result2 = (a, b) => console.log(min(a, b));
 
-test = Boolean(test);
-console.log(test);
+result2(2,10);
+result2(43,19);
+result2(22,22);
 
-// ------------- part 8 ----------------
+result3 = (a, y) => console.log(max(a, y));
 
-let array1 = [2, 3, 5, 8];
-let work = 1;
+result3(2,10);
+result3(43,19);
+result3(22,22);
 
-for (let i = 0; i < array1.length; i++) {
-    work *= array1[i];
-}
 
-console.log(work);
+// ------------- part 8 ----------------
 
-// ------------- part 9 ----------------
 
-let array2 = [2, 5, 8, 15, 0, 6, 20, 3];
+let arr = [12, 80, 14, 01, 16, 0, 18, 19, 2, 70];
 
-for (let i = 0; i < array2.length; i++) {
-    if (array2[i] > 5 && array2[i] < 10)
-        console.log(array2[i]); 
-}
+for (let i = 0; i < arr.length; i++) {
+    arr[i] += "";
 
-// ------------- part 10 ----------------
+    if ((arr[i]).length === 2) {
+        for (let j = 0; j < 2; j++) {
+            
+            if (arr[i].charAt(0) == 0)
+                arr[i] = 'zero' + arr[i].charAt(1);
 
-let array3 = [2, 5, 8, 15, 0, 6, 20, 3];
+            if (arr[i].charAt(1) == 0)
+                arr[i] = arr[i].charAt(0) + 'zero';             
+        }
+    }
 
-for (let i = 0; i < array3.length; i++) {
-    if (array3[i] % 2 == 0)
-        console.log(array3[i]); 
+    if (arr[i] == 0) 
+        arr[i] = 'zero';
 }
+
```

Comment:
It was required to write a function at the task.
And that algorithm doesn't change 0 to zero to the number `100` in the input array. Also, it shouldn't change the type of numbers if they don't include `0`. Look carefully at the task.